### DISCUSSION TITLE
plotIPDF problems

### DISCUSSION THREAD

**Discussion by WongHsai** [mtex-toolbox/mtex#1255] [Ask Anything] [answered]
Hi,community.
       Hope this dicussion got a little help.The IPDF plot I want is the one like this:
![图片](https://user-images.githubusercontent.com/90515523/137049698-74a392ac-fc63-44b3-ae86-94b3bc154f92.png)

The points in IPDF is colorcoded by their orientations in ipfHSVKey based on grains.meanOrientation,and the area is the same scale with the area of specific grains.
However,when I use plotIPDF to achieve that by this:

- cs = loadCIF('Ni-Nickel-cubic.cif');ss = specimenSymmetry('orthorhombic');
- ori=orientation(grains.meanOrientation,cs,ss);
- ipfKey=ipfHSVKey(grains.meanOrientation);
- colors=ipfKey.orientation2color(grains);
- plotIPDF(ori,colors,vector3d.Z,'MarkerSize',0.1*grains.area);

Matlab output:Input argument "r" should be of type "vector3d",
If I restrict the grains.id to a limited number and do that all again:
Another error Matlab output is about the misuse of scatter,S and X are not in the same size or dimension.
Syntax about the position of 'colors',and 'MarkerSize' had been tried a lot but a clear way to do it is not known.

Another wish is about an IPDF with ebsd.orientations on it,with the same color and area demand.So the points would be scattered clouds.
Cheers
Wong Hsai

**Comment by kilir** [comment, marked as answer]
Hi,
as a workaround, you should skip the orthorhombic specimen symmetry and it works fine, at least for me (might need some proper scaling of the blobs though). The plot does not make any difference anyways. What is the reason you'd like to apply orthorhombic specimen symmetry to ebsd data?
Cheers,
Rüdiger

<img width="557" alt="blobs" src="https://user-images.githubusercontent.com/11697557/137093809-1fdf1b1f-466f-4b78-be58-c69e076c3203.png">

**Reply by WongHsai** [reply]
Hi，
Much appreciation for your help.The main reason an orthorhombic specimen symmetry was input is to avoid 
![微信截图_20211013162900](https://user-images.githubusercontent.com/90515523/137096699-8904a890-01fa-42ef-9212-749ddc453d37.png)
 which half of the image is really needed.,like
![微信截图_20211013163354](https://user-images.githubusercontent.com/90515523/137097442-12e26ffb-01ad-4d2a-95f9-4c2d0457dbcb.png)
I learnd this way from scripts shared by others that obviously I found myself even didn't notice it.Is there another way to realize the same result without specimen symmetry?
Cheers
Wong Hsai

**Reply by kilir** [reply]
Does your EBSD system index nickel as '432' ? I thought it would be 'm-3m' ?

**Reply by WongHsai** [reply]
The default setting for my dataset seems to be '432' in the import_wizard.Many thanks for your kind suggestions.Hope this dicussion help more people in need.
